### PR TITLE
Migrate to adonis 5

### PR DESCRIPTION
**Pull request by RodolfoSilva**
- [x] Update project configs to be more like adonis providers
- [x] Make sure the provider works with adonis 5
- [x] Remove `human-interval` 
- [x] Update tests
- [x] Update docs
- [x] Update coverage

### CODE REVIEW

**Comment by RodolfoSilva** [comment]
@HigoRibeiro I don't know why, but the coveralls keep sending this message:

```
Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
```

Can you help me with this? I think you may need to adjust some settings.

**Comment by RodolfoSilva** [comment]
@HigoRibeiro @diego3g @pellizzetti 

I don't know why, but the coveralls keep sending this message:

```bash
yarn run v1.22.5
$ nyc npm test && nyc report --reporter=text-lcov | coveralls
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1599444753514-0.44342638581364535/node but npm is using /opt/hostedtoolcache/node/12.18.3/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> @rocketseat/adonis-bull@0.2.2 pretest /home/runner/work/adonis-bull/adonis-bull
> npm run lint

npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1599444753514-0.44342638581364535/node but npm is using /opt/hostedtoolcache/node/12.18.3/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> @rocketseat/adonis-bull@0.2.2 lint /home/runner/work/adonis-bull/adonis-bull
> eslint . --ext=.ts


> @rocketseat/adonis-bull@0.2.2 test /home/runner/work/adonis-bull/adonis-bull
> node japaFile.js



Provider
  ✓ BullProvider (70ms)

Bull
  ✓ should add a new job (13ms)
  ✓ should add a new job with events inside Job class (1ms)
  ✓ should schedule a new job (9ms)
  ✓ shouldn't schedule when time is invalid (1ms)

 PASSED 

total         : 5
passed        : 5
duration      : 96ms
-----------------------|---------|----------|---------|---------|-------------------
File                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------|---------|----------|---------|---------|-------------------
All files              |   75.81 |    56.52 |      50 |   79.66 |                   
 adonis-bull           |     100 |      100 |     100 |     100 |                   
  japaFile.js          |     100 |      100 |     100 |     100 |                   
 adonis-bull/providers |   90.91 |    66.67 |   66.67 |   90.91 |                   
  BullProvider.ts      |   90.91 |    66.67 |   66.67 |   90.91 | 30                
 adonis-bull/src       |   70.21 |    52.94 |   44.44 |      75 |                   
  BullManager.ts       |   70.21 |    52.94 |   44.44 |      75 | 60-77,97,113      
-----------------------|---------|----------|---------|---------|-------------------

/home/runner/work/adonis-bull/adonis-bull/node_modules/coveralls/bin/coveralls.js:19
      throw err;
      ^
Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
(Use `node --trace-uncaught ...` to show where the exception was thrown)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
##[error]Process completed with exit code 1.
```

Can anyone help me with this? I think may need to adjust some settings.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+[*.json]
+insert_final_newline = ignore
+
+[**.min.js]
+indent_style = ignore
+insert_final_newline = ignore
+
+[MakeFile]
+indent_style = tab
```
Comment (ashokgelal) [inline .editorconfig:23]: I recommend sending formatting/lint related changes in a different PR

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+RUN npm install
+
+RUN wget https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -O /wait-for-it.sh
+RUN chmod +x /wait-for-it.sh
+
+COPY . .
```
Comment (ashokgelal) [inline Dockerfile:17]: Again, I think this doesn't relate to the core purpose of the PR and my recommendation is to send it as a separate PR

File: README.md [hunk 3]
```diff
@@ -9,95 +9,67 @@
...
-    redis: {
-      host: "127.0.0.1",
-      port: 6379,
-      password: null,
-      db: 0,
-      keyPrefix: ""
-    }
-  },
-  remote: "redis://redis.example.com?password=correcthorsebatterystaple"
-};
```
Comment (ashokgelal) [inline README.md:57]: What's the replacement for this? How should one configure connections now?

File: README.md [hunk 4]
```diff
@@ -9,95 +9,67 @@
 
 ## Install
 
-`adonis install @rocketseat/adonis-bull`
+You need [install Redis Provider](https://preview.adonisjs.com/guides/database/redis/#setup) before install this package.
```
Comment (ashokgelal) [inline README.md:12]: Typo: ...before install this package > ...before **installing** this package.

Comment (RodolfoSilva) [reply README.md]: Fixed, thanks! 😊

File: README.md (hunk 3 above)
Comment (RodolfoSilva) [reply README.md:57]: This is intentional, now you should define this following the @adonisjs/redis guide.

Comment (ashokgelal) [reply README.md:57]: Ah! That makes sense. Thanks for clarifying. Maybe adding a link to Adonis Redis page would be helpful. Currently it links to Bull's config page so I was confused.

File: .editorconfig (hunk 1 above)
Comment (RodolfoSilva) [reply .editorconfig]: Done! Thanks.

File: Dockerfile (hunk 2 above)
Comment (RodolfoSilva) [reply Dockerfile]: I agree, I've used this locally. But this is not part of the project.

File: README.md (hunk 3 above)
Comment (RodolfoSilva) [reply README.md]: Done, thanks!

**Comment by zulfi007** [comment]
I tried to complie your code but getting BullBoard  empty.  
 I am getting type error at BullManager.js:72:39: Cannot read property 'listen' of undefined .

**Comment by RodolfoSilva** [comment]
@zulfi007 can you provide a repo with a simple implementation where I can get this error?

**Comment by zulfi007** [comment]
> @zulfi007 can you provide a repo with a simple implementation where I can get this error?

https://github.com/zulfi007/adonis-bull-test.git

also look into typescript typing error. I need to ts-ignore for  import Bull from '@ioc:Rocketseat/Bull'

**Comment by HigoRibeiro** [comment]
First, thanks @RodolfoSilva for your contribution. 

I'll look at this throughout the week. 🚀

**Comment by RodolfoSilva** [comment]
Thanks @HigoRibeiro

**Comment by LeCoupa** [comment]
Do we have an ETA to merge this? 🙂

**Comment by waldandrade** [comment]
@RodolfoSilva falta muito para disponibilizar essa adaptação? rsrsrs

**Comment by RodolfoSilva** [comment]
@waldandrade eu acredito que a **Rocketseet** abandonou esse projeto, ou tem andado bastante ocupada com outras demandas e não tem sobrado tempo para olhar a lib(Mais provável).

Ae agora só eles para te responder mesmo.

**Comment by waldandrade** [comment]
@RodolfoSilva obrigado pela resposta rápida! :D Conhece alguma outra opção de scheduler para AdonisV5?

**Comment by RodolfoSilva** [comment]
@waldandrade eu acabei criando um provider dentro do projeto com o Bull mesmo. Mas vc pode criar um arquivo na pasta start, utilizando o comando `node ace make:prldfile scheduler` e utilizar o node-cron nele se for algo simples.

**Comment by HigoRibeiro** [comment]
Fala galera, beleza?
Cara não abandonamos o projeto não!

Um pouco de várias coisas contribuíram para ainda não dar o merge!!
@RodolfoSilva será que conseguimos conversar no Discord? Higo Ribeiro#5513

Se não for possível eu te mando um e-mail, abraços

**Comment by waldandrade** [comment]
@HigoRibeiro que bom! Espero que dê tudo certo.

**Comment by waldandrade** [comment]
@RodolfoSilva realmente estou precisando utilizar algumas funções coisas mais complexas. Teria como me ensinar como configurar esta lib no Adonis5?

**Comment by RodolfoSilva** [comment]
@waldandrade bom dia, conversei com @HigoRibeiro ontem. Alinhamos algumas coisas, acredito que nos próximos dias teremos uma release. Só preciso conseguir um tempo aqui para fazer esses ajustes.

**Comment by LeCoupa** [comment]
Great @RodolfoSilva waiting for the release :)

**Comment by rubenmoya** [comment]
Is there anything missing to have this merged? How can I help?

**Comment by ammezie** [comment]
Any ETA as to when this can be merged?

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  "extends": "./node_modules/@adonisjs/mrm-preset/_tsconfig",
+  "compilerOptions": {
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
+    "skipLibCheck": true
+  },
+  "files": [
+    "./node_modules/@adonisjs/core/build/adonis-typings/index.d.ts",
+    "./node_modules/@adonisjs/redis/build/adonis-typings/index.d.ts",
```
Comment (ashokgelal) [inline tsconfig.json:11]: Should this be excluded after this? https://github.com/Rocketseat/adonis-bull/pull/39/commits/eaaaecd3c80d080355fc743699e0efa7311a6c36

**Comment by ashokgelal** [comment]
I needed this package in one of the apps I'm working on and published it as a separate npm package as a stop gap solution. If you are in the same situation, please help yourself: https://www.npmjs.com/package/@ashokgelal/adonis-bull

I'll remove the package once this merge gets approved and published. Thanks so much to @RodolfoSilva for this migration. You are the hero :)

**Comment by LeCoupa** [comment]
Thanks @ashokgelal, I will use your package. Any ETA for this PR to be merged? Many companies rely on it. :/

**Comment by RodolfoSilva** [comment]
Thanks @ashokgelal 👍🏽

**Comment by fhugoduarte** [comment]
Hi @RodolfoSilva :hugs: 

First thx for the contrib! We are reviewing this PR, but we have some problems to make it work, as soon as we finish.
We are doing our best to make the merger happen as soon as possible

**Comment by fhugoduarte** [comment]
The work done so far has been very nice!
We need to make some small adjustments to launch the alpha version, we will publish soon.

So, we will need your help to test this new version when it becomes available. You can follow the progress on this issue: #50
